actors/hierarchical_sca/src/exec.rs

```diff
@@ -0,0 +1,197 @@
+use crate::{atomic, StorableMsg};
+
+/// Status of an atomic execution
+#[derive(PartialEq, Eq, Clone, Copy, Debug, Deserialize_repr, Serialize_repr)]
+#[repr(u64)]
+pub enum ExecStatus {
+    /// Oh, oh! something went wrong. This state should never appear.
+    UndefState,
+    /// The atomic execution is initialized and waiting for the submission
+    /// of output states
+    Initialized,
+    /// The execution succeeded.
+    Success,
+    /// The execution was aborted.
+    Aborted,
+}
+
+/// Data persisted in the SCA for the orchestration of atomic executions.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct AtomicExec {
+    pub params: AtomicExecParams,
+    pub submitted: HashMap<String, Cid>,
+    pub status: ExecStatus,
+}
+impl Cbor for AtomicExec {}
+
+impl AtomicExec {
+    pub fn new(params: AtomicExecParams) -> Self {
+        AtomicExec {
+            params,
+            submitted: HashMap::<String, Cid>::new(),
+            status: ExecStatus::Initialized,
+        }
+    }
+}
+
+/// Parameters used to submit the result of an atomic execution.
+#[derive(Clone, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct SubmitExecParams {
+    pub cid: Cid,
+    pub abort: bool,
+    pub output: atomic::SerializedState, // TODO: LockedState
+}
+impl Cbor for SubmitExecParams {}
+
+/// Parameters to uniquely identify and describe an atomic execution.
+///
+/// The unique ID of an execution is determined by the CID of its parameters.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct AtomicExecParams {
+    pub msgs: Vec<StorableMsg>,
+    pub inputs: HashMap<String, LockedStateInfo>,
+}
+impl Cbor for AtomicExecParams {}
+
+/// Output of the initialization of an atomic execution.
+// FIXME: Can we probably return the CID directly without
+// wrapping it in an object (check Go interop)
+#[derive(Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct LockedOutput {
+    pub cid: Cid,
+}
+impl Cbor for LockedOutput {}
+
+/// Output for the submission of an atomic execution.
+/// It returns the state of the atomic execution after the submission.
+#[derive(Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct SubmitOutput {
+    pub status: ExecStatus,
+}
+impl Cbor for SubmitOutput {}
+
+/// Information to identify the locked state from an actor that is running an atomic
+/// execution. To locate some LockedState in a subnet the Cid of the locked state
+/// and the actor where it's been locked needs to be specified.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct LockedStateInfo {
+    pub cid: Cid,
+    pub actor: Address,
+}
+impl Cbor for LockedStateInfo {}
+
+#[derive(PartialEq, Eq, Clone, Serialize_tuple, Deserialize_tuple)]
+pub struct MetaExec {
+    pub msgs_cid: TCid<TAmt<StorableMsg>>,
+    pub input_cid: TCid<THamt<Address, LockedStateInfo>>,
```

Comment:
```suggestion
    pub inputs_cid: TCid<THamt<Address, LockedStateInfo>>,
```
Just because in `AtomicExecParams` it's `msgs` and `inputs`.